File: itlen/app/src/main/java/com/deadely/itl_en/dataclasses/Stat.java
package com.deadely.itl_en.dataclasses;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Stat implements Parcelable {
    @NonNull
    @SerializedName("_id")
    @Expose
    private String id;
    @SerializedName("date")
    @Expose
    private String date;
    @ColumnInfo(name = "stat_words")
    @SerializedName("words")
    @Expose
    private List<Word> words = null;
    @SerializedName("user")
    @Expose
    private User user;

    public Stat() {
    }

    protected Stat(Parcel in) {
        id = in.readString();
        date = in.readString();
        user = in.readParcelable(User.class.getClassLoader());
    }

    public static final Creator<Stat> CREATOR = new Creator<Stat>() {
        @Override
        public Stat createFromParcel(Parcel in) {
            return new Stat(in);
        }

        @Override
        public Stat[] newArray(int size) {
            return new Stat[size];
        }
    };

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public List<Word> getWords() {
        return words;
    }

    public void setWords(List<Word> words) {
        this.words = words;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(id);
        dest.writeString(date);
        dest.writeParcelable(user, flags);
    }
}

File: itlen/app/src/main/java/com/deadely/itl_en/dataclasses/Stat.kt
package com.deadely.itl_en.dataclasses

import android.os.Parcelable
import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.util.*

@Parcelize
@Entity(tableName = "stat_table")
data class Stat(
        @PrimaryKey(autoGenerate = false)
        @ColumnInfo(name = "stat_id")
        @SerializedName("_id") @NonNull val _id: String,
        @SerializedName("date") val date: Date,
        @SerializedName("count_all") val count_all: Int,
        @SerializedName("count_success") val count_success: Int,
        @SerializedName("count_fail") val count_fail: Int
) : Parcelable